src/integration/Sites.spec.ts

```diff
@@ -0,0 +1,205 @@
+import { GitHubService } from "@root/services/db/GitHubService"
+import { ConfigYmlService } from "@root/services/fileServices/YmlFileServices/ConfigYmlService"
+import IsomerAdminsService from "@root/services/identity/IsomerAdminsService"
+import { SitesService } from "@root/services/utilServices/SitesService"
+import { getUsersService } from "@services/identity"
+import { sequelize } from "@tests/database"
+
+const mockSite = "mockSite"
+const mockSiteId = "1"
+const mockBlockedSite = "adminOnly"
+const mockUpdatedAt = "now"
+const mockPermissions = { push: true }
+const mockPrivate = true
+
+const gitHubService = new GitHubService({ axiosInstance: mockAxios })
+const configYmlService = new ConfigYmlService({ gitHubService })
+const usersService = getUsersService(sequelize)
+const isomerAdminsService = new IsomerAdminsService({ repository: IsomerAdmin })
+const sitesService = new SitesService({
+  gitHubService,
+  configYmlService,
+  usersService,
+  isomerAdminsService,
+})
+
+const SitesRouter = new _SitesRouter({ sitesService })
+const sitesSubrouter = SitesRouter.getRouter()
+
+// Set up express with defaults and use the router under test
+const subrouter = express()
+// As we set certain properties on res.locals when the user signs in using github
+// In order to do integration testing, we must expose a middleware
+// that allows us to set this properties also
+subrouter.use((req, res, next) => {
+  const userSessionData = new UserSessionData({
+    isomerUserId: mockIsomerUserId,
+    email: mockEmail,
+  })
+  res.locals.userSessionData = userSessionData
+  next()
+})
+subrouter.use(sitesSubrouter)
+const app = generateRouter(subrouter)
+
+describe("Sites Router", () => {
+  afterEach(() => {
+    jest.resetAllMocks()
+    mockAxios.reset()
+  })
+
+  describe("/", () => {
+    beforeAll(async () => {
+      // Set up User and Site table entries
+      await User.create({
+        id: mockIsomerUserId,
+      })
+      await Site.create({
+        id: mockSiteId,
+        name: mockSite,
+        apiTokenName: "token",
+        jobStatus: "READY",
+        siteStatus: "LAUNCHED",
+        creatorId: mockIsomerUserId,
+      })
+      await Site.create({
+        id: "200",
+        name: mockBlockedSite,
+        apiTokenName: "token",
+        jobStatus: "READY",
+        siteStatus: "LAUNCHED",
+        creatorId: mockIsomerUserId,
+      })
+      await SiteMember.create({
+        userId: mockIsomerUserId,
+        siteId: mockSiteId,
+        role: "ADMIN",
+      })
+      await Repo.create({
+        name: mockSite,
+        url: "url",
+        siteId: mockSiteId,
+      })
+    })
+    afterEach(async () => {
+      // Clean up so that different tests using
+      // the same mock user don't interfere with each other
+      await IsomerAdmin.destroy({
+        where: { userId: mockIsomerUserId },
+      })
+    })
+    it("should return list of only sites available to user", async () => {
```

Comment:
nit: we should specify that this is an email user as github users have slightly diff behaviour

Reply:
Fixed in https://github.com/isomerpages/isomercms-backend/pull/480/commits/1a88af565617b2f51cb3932b664c811902078c4c!